src/config.rs

```diff
@@ -120,6 +139,62 @@ impl Default for Config {
 			trusted_peers_0conf: Vec::new(),
 			probing_liquidity_limit_multiplier: DEFAULT_PROBING_LIQUIDITY_LIMIT_MULTIPLIER,
 			log_level: DEFAULT_LOG_LEVEL,
+			anchor_channels_config: Some(AnchorChannelsConfig::default()),
+		}
+	}
+}
+
+/// Configuration options pertaining to 'Anchor' channels, i.e., channels for which the
+/// `option_anchors_zero_fee_htlc_tx` channel type is negotiated.
+///
+/// Prior to the introduction of Anchor channels, the on-chain fees paying for the transactions
+/// issued on channel closure were pre-determined and locked-in at the time of the channel
+/// opening. This required to estimate what fee rate would be sufficient to still have the
+/// closing transactions be spendable on-chain (i.e., not be considered dust). This legacy
+/// design of pre-anchor channels proved inadequate in the unpredictable, often turbulent, fee
+/// markets we experience today. In contrast, Anchor channels allow to determine an adequate
+/// fee rate *at the time of channel closure*, making them much more robust in the face of fee
+/// spikes. In turn, they require to maintain a reserve of on-chain funds to be able to get the
+/// channel closure transactions confirmed on-chain, at least if the channel counterparty can't
+/// be trusted to this for us.
+///
+/// See [BOLT 3] for more technical details on Anchor channels.
+///
+///
+/// ### Defaults
+///
+/// | Parameter                  | Value  |
+/// |----------------------------|--------|
+/// | `trusted_peers_no_reserve` | []     |
+/// | `per_channel_reserve_sats` | 25000  |
+///
+///
+/// [BOLT 3]: https://github.com/lightning/bolts/blob/master/03-transactions.md#htlc-timeout-and-htlc-success-transactions
+#[derive(Debug, Clone)]
+pub struct AnchorChannelsConfig {
+	/// A list of peers which we trust to spend the anchor output *for us* on channel closing.
+	///
+	/// Channels with these peers won't count towards the retained on-chain reserve and we won't
+	/// take any action to get the Anchor spending transactions confirmed.
+	///
+	/// **Note:** Trusting the channel counterparty to spend the anchor output is potentially
+	/// insecure as the channel may not be closed if they refuse to do so, potentially leaving the
+	/// user funds stuck.
+	pub trusted_peers_no_reserve: Vec<PublicKey>,
+	/// The amount of satoshis we keep as an emergency reserve in our on-chain wallet in order to
+	/// be able to spend the Anchor output on channel close.
+	///
+	/// **Note:** Depending on the fee market at the time of closure, this reserve amount might or
+	/// might not suffice to successfully spend the Anchor output, i.e., you may want to adjust
+	/// this value accordingly.
+	pub per_channel_reserve_sats: u64,
```

Comment:
Yeah, I can see why some technical details are necessary to expose, but at least the _implementation_ details (i.e., anchors) can be hidden it seems. Other than in the config, it is only exposed in `ChannelType::Anchors`. And the alternative, `ChannelType::StaticRemoteKey` is not providing any useful information because anchor channels imply that.

Seems instead of `ChannelType` we want something like `ReserveType`. And that can be more fine-grained than two variants as the trusted peer case could be an additional variant. That would provide more useful information as otherwise the user needs to cross reference `ChannelDetails` with the config to see if the counterparty is in `trusted_peers_no_reserve`.

Reply:
Was thinking more along the lines `Predetermined`, `Dynamic`, `NoReserve`. Probably could iterate on the exact names, but basically something that indicates how the reserve is determined.

Reply:
Maybe not necessarily `Predetermined` but something indicating that it is part of the channel balance. Whereas the other is from on-chain funds.

Reply:
> Tbh. I'm not entirely convinced we should go this way. The main reason why I introduced `ChannelType` is for users to check whether any opened channel is or isn't using Anchors, as this has several other consequences other than reserve. E.g., LND requires to use Anchor channels for zero-conf. So I'm afraid entirely dropping `Anchors` from the type obfuscates things even further.

Not sure I understand why the user needs to check this and what actions they would take. Could this be documented?

Reply:
Yeah, either just drop it or use a bool. No need to define a type, IMO. My earlier thought was really on how to tie the reserve configuration to each channel, which might be valuable? Don't really have a strong opinion though.